File: src/web/WikipediaQBuilder.java
package web;

import java.net.URI;
import util.parsing.ParsesFromString;
import util.parsing.StringParseStrategy;
import util.parsing.StringParseStrategy.From;
import util.plugin.IsPlugin;
import static util.Util.urlEncodeUtf8;
import static util.parsing.StringParseStrategy.To.CONSTANT;

@IsPlugin
@StringParseStrategy(from = From.ANNOTATED_METHOD, to = CONSTANT, constant = "Wikipedia")
public class WikipediaQBuilder implements SearchUriBuilder {

	@ParsesFromString
	public WikipediaQBuilder() {}

	@Override
	public URI apply(String q) {
		return URI.create("https://en.wikipedia.org/wiki/" + urlEncodeUtf8(q));
	}

}

File: src/web/WikipediaQBuilder.kt
package web

import util.parsing.StringParseStrategy
import util.parsing.StringParseStrategy.From
import util.parsing.StringParseStrategy.To
import java.net.URI

@StringParseStrategy(from = From.SINGLETON, to = To.CONSTANT, constant = "Wikipedia")
object WikipediaQBuilder: WebSearchUriBuilder {

    override fun doApply(q: String) = URI.create("https://en.wikipedia.org/wiki/$q")!!

}